src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -273,6 +273,10 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 							shellLaunchConfig.env = shellLaunchConfig.env || {} as IProcessEnvironment;
 							shellLaunchConfig.env['VSCODE_SHELL_LOGIN'] = '1';
 						}
+						if (env?.['ZDOTDIR']) {
```

Comment:
I think `?.` for here and above isn't actually needed anymore, we should also remove the TODO: fix above

Reply:
We may also be able to remove these all together as env is passed into `createProcess`? See how `VSCODE_SHELL_HIDE_WELCOME` doesn't need to get set on `shellLaunchConfig.env`

Reply:
we need to resolve the execInstallFolder there for the ZDOTDIR one
and see this TODO - slc.env is used, not env in remote terminal backend https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/terminal/browser/remoteTerminalBackend.ts#L161